### PR TITLE
feat(rpm_repository): add retain_repo_versions parameter

### PR DESCRIPTION
**Pull request by joey-grant**
This change adds `retain_repo_versions` parameter option to `rpm_repository` module.

### CODE REVIEW

**Comment by joey-grant** [comment]
This is related to https://github.com/pulp/squeezer/issues/165

File: plugins/modules/rpm_repository.py [hunk 1]
```diff
@@ -41,6 +41,10 @@
       - Max number of latest versions of each package to keep (optional)
     type: int
     version_added: "0.0.16"
+  retain_repo_versions:
+    description:
+      - Max number of repository versions to keep (optional)
```
Comment (mdellweg) [inline plugins/modules/rpm_repository.py]:
```suggestion
      - Max number of repository versions to keep
```
not being required is implied by ansible. Can you adjust line 41 too?

Also version_added may be nice.

**Review by mdellweg — COMMENTED** [review]
How do you feel about adding tests?
These modules are about to be transfered to a new framework. Adding this parameter to the tests will ensure it's not immediately broken afterwards.
(Please do not test the retain version feature, but just that the model is able to set it on a repository.)

File: plugins/modules/rpm_repository.py (hunk 1 above)
Comment (joey-grant) [reply plugins/modules/rpm_repository.py]: Thanks for the feedback, @mdellweg . I've made the adjustments and will submit them soon. 

Can you confirm what you would like to see regarding tests? I have produced 2 tests:

1. set `retain_repo_versions` to 5
2. update `retain_repo_versions` to 1 (i.e. don't keep old versions)

I have ran and recorded the tests--they all pass. However, I see the following now:

1. all existing `tests/fixtures/rpm_repository-*.yml` files have been modified
2. 2 new fixture files were created: `test/fixtures/rpm_repository-{22,23}.yml`

What portions of the tests/results would you like committed?

Side note:
There seems to be a limitation in which every call that includes the `retain_repo_versions` parameter updates the returned `pulp_last_updated` timestamp (no other returned values differ). This is the case even when `retain_repo_versions` is not changed from the previous call. I assume this is a bug in `pulpcore` itself; however, it does limit the ability to assert `results.changed == false` on repeated calls with the same value for `retain_repo_versions`.

Comment (mdellweg) [reply plugins/modules/rpm_repository.py]: > What portions of the tests/results would you like committed?

All of them. They are separate files for technical reasons only.

> Side note: [...]

That actually sounds like a bug on the client side. When we don't want to change the value, we must not do a patch call, and that still seems to happen.

Comment (joey-grant) [reply plugins/modules/rpm_repository.py]: Hey @mdellweg, I have changed the tests to try for idempotence and am not seeing the previous issue. I believe that I must have made an error in my original work. Please see my changes and let me know if there is anything else needed for approval.

I appreciate you allowing me to contribute to this project.

**Review by mdellweg — DISMISSED** [review]
This looks good.
Thank you!

**Comment by mdellweg** [comment]
Looks like you need to run "make format" once.

**Comment by joey-grant** [comment]
> Looks like you need to run "make format" once.

Hey @mdellweg, I've pushed the newly formatted code.

**Review by mdellweg — APPROVED** [review]
Thank you for following along.

**Pull request merged**